### PR TITLE
fix: autocommit re-tries failed DB tx

### PR DESCRIPTION
**Pull request by elsirion**
We got a comparison the wrong way around, this commit adds a test so it doesn't break again and fixes the issue.

### CODE REVIEW

**Review by dpc — APPROVED** [review]

File: fedimint-api/src/db/mod.rs [hunk 1]
```diff
@@ -709,9 +709,17 @@ impl DecodingError {
 }
 
 mod tests {
```
Comment (dpc) [inline fedimint-api/src/db/mod.rs:711]: `#[cfg(tests)]` missing (test? or tests?)

Comment (elsirion) [reply fedimint-api/src/db/mod.rs]: It's not missing on the `mod`, maybe I should create a separate `mod` though. The problem is the existing test `fn`s aren't standalone, but just a template to test all other database implementatios.

**Comment by elsirion** [comment]
Sorry for pulling you in @justinmoon, I think Intellij had a bug and added you as a commit author :laughing:

File: fedimint-api/src/db/mod.rs (hunk 1 above)
Comment (elsirion) [reply fedimint-api/src/db/mod.rs]: Clippy is happy after I put the relevant `use`s into the test `fn`.

Comment (dpc) [reply fedimint-api/src/db/mod.rs]: I'm not sure if I get it. Should we rename this module if it's not a module with tests? `dbtest` or `testutil`? All other `mod tests` are exclusively for tests.

**Review by dpc — APPROVED** [review]

**Comment by mxz42** [comment]
oh, wow. my bad. great that it's fixed and that there's a test now

**Comment by justinmoon** [comment]
@elsirion I will take credit for your work any time!

**Pull request merged**